spec/integration/rspec/unhandled_matcher_spec.rb

```diff
@@ -0,0 +1,86 @@
+require "spec_helper"
+
+RSpec.describe "Integration with built-in RSpec matchers", type: :integration do
+  context "when the expected value contains a built-in matcher" do
+    context "that fails" do
+      it "produces the correct failure message when used in the negative" do
+        as_both_colored_and_uncolored do |color_enabled|
+          snippet = <<~TEST.strip
+            expected = hash_including(
+              number: be_a(Numeric)
+            )
+            actual = {
+              number: 4
+            }
+            expect(actual).not_to match(expected)
+          TEST
+          program =
+            make_plain_test_program(snippet, color_enabled: color_enabled)
+
+          expected_output =
+            build_expected_output(
+              color_enabled: color_enabled,
+              snippet: "expect(actual).not_to match(expected)",
+              expectation:
+                proc do
+                  line do
+                    plain "Expected "
+                    actual "{ number: 4 }"
+                    plain " not to match "
+                    expected "#<a hash including (number: (be a kind of Numeric))>"
```

Comment:
Hmm, so now we have a mixture of syntaxes. We have the `#<...>` style, which is used for `hash_including` (and other similar "fuzzy objects") and `(...)` for the arguments to the fuzzy object. Should we also use `#<...>` for `be_a`?

Reply:
Ah, I couldn't quite glean what opening/closing symbols signified, but now I see – makes sense, thanks! Updated.